src/Agoda.IoC.NetCore/StartupExtension.cs

```diff
@@ -8,31 +8,49 @@
 
 namespace Agoda.IoC.NetCore
 {
+    public class KeyTypePair
+    {
+        public Type Type { get; set; }
+        public object Key { get; set; }
```

Comment:
what is the decision behind using `Object key` and not use `String key`? is there any use case for non-string key? so we can avoid boxing unboxing.

Reply:
It's a good question, the original framework done in unit was using object as a key, so in order to make it work with the original code i needed to use object, we should probably move to string. But unsure what the impact is. I'll have a look

Reply:
I've updated it to string, its a breaking change to the unity implementation, but i doubt anyone is using it, the unity implementation was mainly published for historical reasons.